### PR TITLE
Paginate feed md

### PR DESCRIPTION
**Pull request by dmorina**
adds pagination to the feed
random 2 to 6 seconds delay

only for worker pool 2 (boomerang)
addresses #521

### CODE REVIEW

File: static/js/task-feed/controllers/task-feed.controller.js [hunk 1]
```diff
@@ -1,130 +1,177 @@
...
+        self.openComments = openComments;
+        self.saveComment = saveComment;
+        self.loading = true;
+        self.getStatusName = getStatusName;
+        self.currentPage = 1;
+        self.page_data = {};
+        self.page_numbers = [];
+        self.switchPage = switchPage;
+        self.getCollection = getCollection;
+        self.has_pagination = has_pagination;
```
Comment (dmorina) [inline static/js/task-feed/controllers/task-feed.controller.js]: some methods and variables added here

File: static/js/task-feed/controllers/task-feed.controller.js [hunk 2]
```diff
@@ -1,130 +1,177 @@
...
+                self.availableTasks = false;
+                var k = 0;
+                var page = [];
+                //experiment code -- pagination
+                for (var i = 0; i < self.projects.length; i++) {
+                    page.push(self.projects[i]);
+                    if (((i + 1) % 5) == 0 || self.projects.length - i == 1) {
+                        k++;
+                        self.page_data[k] = page;
+                        page = [];
```
Comment (dmorina) [inline static/js/task-feed/controllers/task-feed.controller.js]: paginates the results 5 per page

File: static/js/task-feed/controllers/task-feed.controller.js [hunk 3]
```diff
@@ -1,130 +1,177 @@
...
+            else return 'Running';
+        }
+
+        function switchPage(index) {
+            var random = Math.floor(Math.random() * (5)) + 2;
+            $mdToast.showSimple('Loading...');
+            setTimeout(function () {
+                self.currentPage = index;
+                $scope.$apply();
+            }, random*1000);
```
Comment (dmorina) [inline static/js/task-feed/controllers/task-feed.controller.js]: switches page in 2 to 6 random seconds

File: static/js/task-feed/controllers/task-feed.controller.js [hunk 4]
```diff
@@ -1,130 +1,177 @@
...
+        }
+
+        function getCollection(){
+            if(has_pagination()){
+                return self.page_data[self.currentPage];
```
Comment (dmorina) [inline static/js/task-feed/controllers/task-feed.controller.js]: checks which data array to return

File: static/templates/task-feed/main.html [hunk 5]
```diff
@@ -109,5 +112,16 @@ <h2 class="md-title">No free tasks available</h2>
...
     </ng-include>-->
+</div>
+<div ng-if="taskfeed.has_pagination()" class="pagination" layout="row" layout-align="center center">
+    <div layout="column" flex="60">
+        <div layout="row" layout-align="center center">
+            <md-button ng-class="{'active': i==taskfeed.currentPage}" ng-repeat="i in taskfeed.page_numbers"
+                       class="pagination-button"
+                       ng-click="taskfeed.switchPage(i)">
+                {{ i }}
+            </md-button>
```
Comment (dmorina) [inline static/templates/task-feed/main.html]: html stuff :D

**Comment by aginzberg** [comment]
lgtm

**Comment by mbernst** [comment]
Six seconds seems a bit high. Can we do 2-4sec? This is based on what
Google found:
http://www.slideshare.net/dyninc/the-user-and-business-impact-of-server-delays-additional-bytes-and-http-chunking-in-web-search-presentation
slide 4

**Comment by dmorina** [comment]
sure thing

**Comment by dmorina** [comment]
patching directly ...

**Comment by dmorina** [comment]
done

**Pull request merged**